scripts/20_save_data.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/sh
+#
+# Copyright (C) 2018 Ultimaker B.V.
+# Copyright (C) 2018 Olliver Schinagl <oliver@schinagl.nl>
+#
+# SPDX-License-Identifier: AGPL-3.0+
+
+set -eu
+
+trap cleanup EXIT
+
+# system_update wide configuration settings
+SYSTEM_UPDATE_CONF_DIR="${SYSTEM_UPDATE_CONF_DIR:-}"
+TARGET_STORAGE_DEVICE="${TARGET_STORAGE_DEVICE:-}"
+UPDATE_ROOTFS_SOURCE="${UPDATE_ROOTFS_SOURCE:-}"
+UPDATE_SAVE_DIR="${UPDATE_SAVE_DIR:-}"
+# end system_update wide configuration settings
+
+KEEP_LIST="$(mktemp)"
+
+NAME_TEMPLATE_TARGET_MOUNT="um-save_mount"
+TARGET_MOUNT="$(mktemp -d -t "${NAME_TEMPLATE_TARGET_MOUNT}.XXXXXX")"
+
+usage()
+{
+    echo "Usage: ${0} [OPTIONS]"
+    echo "Save the data files from the TARGET_STORAGE_DEVICE."
```

Comment:
Well you could say all files are data, but that means that everything is saved from the TARGET_STORAGE_DEVICE. Suggest: Save (Why not backup) files listed in the .keep files found in the <SYSTEM_CONFIG_DIR>